### PR TITLE
[5228] add save and sync db using pouchdb

### PR DESCRIPTION
**Pull request by ilmiawan**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
Need to add capability of saving request data using pouchdb and sync it to external couchdb

## How did you implement / how did you fix it  
1.  Add pouchdb to save requests and notifications result
2. Add synchronize ability between pouchdb and external couchdb

## How to test  
1. run monika in symon mode with the flag `symonExperimental` 
2. add external couchdb url using  `symonCouchDbURL` flag
3. running monika command example :
```
npm start -- --symonUrl http://localhost:3000 --symonKey 7e73179e-74f0-4cd2-91ad-85b4f346b221 --verbose --symonCouchDb http://symon:symon@localhost:5984/symon 
```

### CODE REVIEW

File: src/symon/index.ts [hunk 1]
```diff
@@ -256,6 +265,17 @@ class SymonClient {
...
+    const { flags } = getContext()
+
+    this.isSymonExperimental = flags.symonExperimental
+
+    this.symonCouchDB =
+      flags.symonCouchDbURL || 'http://symon:symon@localhost:5984/symon'
+
+    this.pouch = new PouchDB('symon')
+
+    this.remotePouchDB = new PouchDB(this.symonCouchDB)
```
Comment (nicnocquee) [inline src/symon/index.ts:278]: the variable naming is a bit confusing. the flag is `symonCouchDBURL`, assigned to variable called `symonCouchDB`, then used to initiate a Pouch DB. 

Can you rename it so they make more sense?

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/symon/index.ts (hunk 1 above)
Comment (ilmiawan) [reply src/symon/index.ts]: done renaming to symonCouchDB

File: src/symon/index.ts [hunk 2]
```diff
@@ -151,14 +151,22 @@ class SymonClient {
...
     logger: log,
     doRootCheck: false,
     errorHandler: (error, workerMetadata) => {
       log.error(error)
       log.debug(workerMetadata)
     },
+    removeCompleted: true,
     outputWorkerMetadata: true,
+    workerMessageHandler: async ({ message }) => {
+      console.log(message)
```
Comment (lukman7788) [inline src/symon/index.ts:163]: @ilmiawan please remove debug code

**Comment by ilmiawan** [comment]
done experiment with couchdb.

**Pull request closed without merging**